### ISSUE TITLE
Whitelabeling for other users

### ISSUE THREAD

**Issue by thisisparker** [thisisparker/foiafeed#3]
There are currently a few hardcoded references to the name FOIAFeed and the specific nature of the project in the source code. (Off the top of my head, the phrases list, the database name, and the user-agent all qualify.)

Abstracting some of those out into a config file or similar would help other users adopt this framework to run their own notifications-style bots on other topics or pulling from different input sources or RSS feeds.

**Comment by thisisparker** [comment]
A few recent changes might have brought this to the right balance; the database name is set in a config file, and most or all of the "configurables" that are hardcoded in are set up top in the script, which will make reconfiguring it pretty trivial. It might still be worth pulling them out into a config, depending on whether the best model for other users of this project is to fork their own repo or contribute directly to this one.

**Comment by thisisparker** [comment]
Actually, @jfilter, do you have an opinion on the above? If you are still interested in applying this project to a German context, would you rather you fork it and run its own repo, or for me to pull out all the settings needed for porting it into config files you could create?

**Comment by jfilter** [comment]
Hey Parker, here is my take on it:

a) You have build a tool to monitor websites via RSS feeds for specific words. There is default processing for each domain, but also need to be able to write custom handlers. The tool returns the URL and the text surrounding that specific word.

b) You have build a tool to generate images from text and tweet them.

It would be cool two separate your code into two modules. In addition a) should allow users to create custom handlers as part of the config. And b) can be replaced with modules that post it on Facebook, toot it etc.. I don't know how much time you want to spend on it. Your code works find for your specific use case. I can take a look into this. ;) 

Currently, I can't estimate how much time and work it will take me to adapt it Germany. There are way more smaller regional newspapers with an horrible IT infrastructure. I assume it will take a lot of fine-tuning so I would like to fork your code.

Let me know your thoughts!
Johannes

**Comment by thisisparker** [comment]
This makes a lot of sense, @jfilter. I'm going to create a new issue and work on splitting up the current script into two modules. That should be relatively quick, and will probably constitute an MVP for "release" (which, as I've mentioned, will be over on the Freedom of the Press page). Extending (A) to include custom handlers and non-RSS sources will be a future feature, and so will extending (B) to include toots and Facebooks and Peach posts and Slack integrations, etc, etc.

Do you happen to know if `readability` works on pages written in German? I think it might take into consideration CSS class names and the like... forgive my ignorance, but are those class names on German news sites likely to be in English, or no?

**Comment by jfilter** [comment]
Yes, most of the class names are in English. If not, they are ASCII compliant. `readability` should work. ;)

**Comment by jfilter** [comment]
I just came across this python tool that does the crawling: https://github.com/fhamborg/news-please

**Comment by thisisparker** [comment]
wow @jfilter that seems like a good tool, and I hadn't seen it! I'm very close to shipping a version based on `readability` but I will be evaluating whether it makes sense to incorporate that into future versions.
